### ISSUE TITLE
Pagination can remove user reaction from bot?

### ISSUE THREAD

**Issue by nathan130200** [DSharpPlus/DSharpPlus#305]
Just a small thing, can bot remove user reaction when switch message in paginated messages? And if don't have permission don't remove reaction 🤔

**Comment by uwx** [comment]
Doesn't it already do that?

**Comment by nathan130200** [comment]
Only in D#+ 4.0 right?

**Comment by NaamloosDT** [comment]
I'll consider adding it as an option. Reaction rate limits are fucked up so its better to just listen to the events. I can't promise it'll make kt in.

**Comment by NaamloosDT** [comment]
I'm sorry, but I can't add this in. Discord gets really goofy when you remove reactions right after adding them, and exceptions will be thrown if the user is faster than the bot (which is very possible taking into consideration that rate limits are a thing)

**Comment by Emzi0767** [comment]
@nathan130200 in its initial revisions, Interactivity immediately removed user reactions. However this proved unfeasible, not because of ratelimits, but because in majority of the cases, the bots lacked permissions necessary to remove said reactions, which caused exceptions to be thrown, methods to be interrupted, and due to how interactivity works internally, this could lead to potential memory leaks.

While the memory leaks were addressed later, the issue of permissions remains. We eventually decided that removing the reactions is unviable, and that Interactivity was going to use reaction add/remove events instead.
